src/pgp/gpg.c

```diff
@@ -769,6 +750,200 @@ p_gpg_format_fp_str(char* fp)
+ *              If the key ID is empty or NULL, returns NULL.
+ * @return The public key data as a null-terminated char* string allocated using malloc.
+ *         The returned string should be freed by the caller.
+ *         Returns NULL on error, and errors are written to the error log.
+ */
+char*
+p_gpg_get_pubkey(const char* keyid)
+{
+    if (!keyid || *keyid == '\0') {
+        return NULL;
+    }
+
+    gpgme_ctx_t context = NULL;
+    gpgme_error_t error = GPG_ERR_NO_ERROR;
+    gpgme_data_t data = NULL;
+
+    error = gpgme_new(&context);
+    if (error != GPG_ERR_NO_ERROR) {
+        log_error("GPG: Failed to create gpgme context. %s", gpgme_strerror(error));
+        goto cleanup;
+    }
+
+    error = gpgme_data_new(&data);
+    if (error != GPG_ERR_NO_ERROR || data == NULL) {
+        log_error("GPG: Failed to create new gpgme data. %s", gpgme_strerror(error));
+        goto cleanup;
+    }
+
+    gpgme_set_armor(context, 1);
+
+    error = gpgme_op_export(context, keyid, GPGME_EXPORT_MODE_MINIMAL, data);
+    if (error != GPG_ERR_NO_ERROR) {
+        log_error("GPG: Failed to export public key. %s", gpgme_strerror(error));
+        goto cleanup;
+    }
+
+cleanup:
+    gpgme_release(context);
+    return _gpgme_data_to_char(data);
+}
+
+/**
+ * Validate that the provided buffer has the format of an armored public key.
+ *
+ * This function only briefly checks for presence of armored header and footer.
+ *
+ * @param buffer The buffer containing the key data.
+ * @return TRUE if the buffer has the expected header and footer of an armored public key, FALSE otherwise.
+ */
+gboolean
+p_gpg_is_public_key_format(const char* buffer)
+{
+    if (buffer == NULL || buffer[0] == '\0') {
+        return false;
+    }
+
+    const char* headerPos = strstr(buffer, PGP_PUBLIC_KEY_HEADER);
+    if (headerPos == NULL) {
+        return false;
+    }
+
+    const char* footerPos = strstr(buffer, PGP_PUBLIC_KEY_FOOTER);
+
+    return (footerPos != NULL && footerPos > headerPos);
+}
+
+/**
+ * Imports a PGP public key(s) from a buffer.
+ *
+ * @param buffer The buffer containing the PGP key data.
+ * @return A pointer to the first imported ProfPGPKey structure, or NULL if an error occurs.
+ *
+ * @note The caller is responsible for freeing the memory of the returned ProfPGPKey structure
+ *       by calling p_gpg_free_key() to avoid resource leaks.
+ */
+ProfPGPKey*
+p_gpg_import_pubkey(const char* buffer)
+{
+    gpgme_ctx_t ctx;
+    ProfPGPKey* result = NULL;
+    gpgme_error_t error = gpgme_new(&ctx);
+    if (error != GPG_ERR_NO_ERROR) {
+        log_error("GPG: Error creating GPGME context");
+        goto out;
+    }
+
+    gpgme_data_t key_data;
+    error = gpgme_data_new_from_mem(&key_data, buffer, strlen(buffer), 1);
+    if (error != GPG_ERR_NO_ERROR) {
+        log_error("GPG: Error creating GPGME data from buffer");
+        goto out;
+    }
+
+    gpgme_import_result_t import_result;
```

Comment:
This declaration could/should should also go down to where it's actually used first.